src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -40,7 +40,7 @@
      */
     public function __construct($name, NodeInterface $parent = null)
     {
-        if (false !== strpos($name, '.')) {
+        if (false !== strpos($name = (string) $name, '.')) {
```

Comment:
`$name` must be a string, so casting here should be removed. Adding this is not a bug fix.

Reply:
in 4.0 it's `string|null` already

https://github.com/symfony/symfony/blob/d65c43ba843c84105c194d9de03945b9cf5cee79/src/Symfony/Component/Config/Definition/BaseNode.php#L42

hence #26297

practically using `null` is just an _alias_ for empty string, but we use null by design: 

https://github.com/symfony/symfony/blob/d65c43ba843c84105c194d9de03945b9cf5cee79/src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php#L75-L78

if we preserve `null` on the property side (thus no cast) we keep violating `getName`:

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Config/Definition/BaseNode.php#L223-L225

hence the cast